frontend/src/pages/ActivityReport/Pages/components/goalValidator.js

```diff
@@ -0,0 +1,32 @@
+export const UNFINISHED_OBJECTIVES = 'Every objective must have both a title and TTA provided';
+export const GOALS_EMPTY = 'Every report must have at least one goal';
+
+export const unfinishedObjectives = (objectives) => {
+  // Every objective for this goal has to have a title and ttaProvided
+  const unfinished = objectives.some(
+    (objective) => !(objective.title && objective.ttaProvided),
+  );
+  return unfinished ? UNFINISHED_OBJECTIVES : false;
+};
+
+export const unfinishedGoals = (goals) => {
+  for (let i = 0; i < goals.length; i += 1) {
+    const goal = goals[i];
+    // Every goal must have an objective for the `goals` field has unfinished goals
```

Comment:
I think this comment is lying. Goals with no objectives are currently valid in the form. I think we do want to enforce that every goal does have at least one objective though, so we should update the code and test line 142 to make the comment accurate

Reply:
I started with the assumption that goals must have at least one objective but during review that changed. I can switch back to requiring every goal have at least one objective. Doesn't matter for this ticket but does every goal on a report imported from smartsheets also have at least one objective?

Reply:
The smartsheet report has a goals field and an objectives field, but it'll be difficult to accurately map which objective maps to which goal.

I don't think it makes sense to have a goal with no objective, since the objective is how any forward progress is made and described